model/Infrastructure/QtiItemResponseValidator.php

```diff
@@ -113,4 +120,34 @@ private function getAllowSkip(AssessmentTestSession $testSession): bool
             ->getItemSessionControl()
             ->doesAllowSkipping();
     }
+
+    /**
+     * Detect whether SCORE or MAXSCORE outcomes are declared.
+     *
+     * @param mixed $outcomeDecls Iterable collection of outcome declarations
+     * @return array{0: bool, 1: bool} [hasScore, hasMaxScore]
+     */
+    private function detectScoringOutcomes($outcomeDecls): array
+    {
+        $hasScore = false;
+        $hasMaxScore = false;
+
+        if (is_iterable($outcomeDecls)) {
+            foreach ($outcomeDecls as $outcomeDecl) {
+                if (is_object($outcomeDecl) && method_exists($outcomeDecl, 'getIdentifier')) {
+                    $identifier = strtoupper((string) $outcomeDecl->getIdentifier());
+                    if ($identifier === 'SCORE') {
+                        $hasScore = true;
+                    } elseif ($identifier === 'MAXSCORE') {
+                        $hasMaxScore = true;
+                    }
+                }
+            }
+        } elseif (is_object($outcomeDecls) && method_exists($outcomeDecls, 'count') && $outcomeDecls->count() === 0) {
```

Comment:
nitpick: you already have fallback to [false, false] anyway, and in the case with empty collection it will return the same, what is a point?